### PR TITLE
Rewrite README to be consistent with other apps

### PR DESCRIPTION
**Pull request by benthorner**
https://trello.com/c/Psf7EzPB/204-move-away-from-startupsh

This implements changes to how this app is run locally, as agreed
in [1]. This also makes other changes to the README to make it more
concise, correct and actionable, which is consistent to changes made
in other READMEs across other, non-frontend repos on GOV.UK.

This PR is meant to establish conventions for changes to other frontend
repos, and should be used as a reference for future changes.

Please see the commit messages for more details.

[1]: https://github.com/alphagov/govuk-developer-docs/pull/2897

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: README.md
Comment (ChrisBAshton) [inline README.md:121]: I don't think it would hurt to copy this info into the newly created taxonomy-pages.md.

File: docs/campaign-pages.md
Comment (ChrisBAshton) [inline docs/campaign-pages.md:5]: Not your fault, but could these links be normalised? There are a few oddities in this markdown.

```suggestion
## Transition landing page ([/transition](https://www.gov.uk/transition))
```

File: README.md
Comment (ChrisBAshton) [inline README.md:132]: Good spot 👍

File: README.md
Comment (ChrisBAshton) [inline README.md:59]: Here's a new a11y suggestion from me: we now have two links with text as "the guidance", which is better than just "here" or "read more", but is still [quite vague when consumed in isolation](https://fae.disability.illinois.edu/rulesets/LINK_2/). This might be better:

```suggestion
Read [how to run and modify Pact tests](https://docs.publishing.service.gov.uk/manual/pact-broker.html).
```

File: README.md
Comment (ChrisBAshton) [inline README.md:79]: Hmmm I did have some misgivings about removing this, as it could be quite useful to have an awareness of the different types of browse pages. But agreed it can be discovered at point of need. And the [vocabulary is already in the code](https://github.com/alphagov/collections/blob/c63209674bae27007a08ea172e7befb56fe500b5/config/routes.rb#L22-L24), so we're not particularly losing anything here.

**Review by ChrisBAshton — APPROVED** [review]
LGTM 👍 some non-blocking suggestions below.

File: README.md
Comment (benthorner) [reply README.md:121]: Sure, I agree it's got slightly different information in it than the separate doc. I've added a new commit to incorporate it.

File: docs/campaign-pages.md
Comment (benthorner) [reply docs/campaign-pages.md]: That's weird, and yes why not. I've amended in 39b40d4f78a7eb44228fa25272db86c6b5aca33b

File: README.md
Comment (benthorner) [reply README.md:59]: That's fair, particularly for the tech docs intro, which I'm expecting to be copied in future PRs for other frontend apps. I've amended both in the relevant commits.

How does it look now?

**Review by ChrisBAshton — APPROVED** [review]
Looks good 👍 thanks.

**Comment by benthorner** [comment]
I've moved two of the commits into a separate PR (without changes): https://github.com/alphagov/collections/pull/2375. This is to make it easier to get a second review from a frontend dev. I'll merge this one now, since the commits that remain don't set any new precedents.

**Comment by benthorner** [comment]
> I'll merge this one now

No I won't, because I can't 😄. @ChrisBAshton would you like to press the button?

**Pull request merged**
